src/features/utils/lazy-loader.js

```diff
@@ -11,27 +31,27 @@ import { FEATURE_NAMES } from '../../loaders/features/features'
  * @param featurePart Name of the feature part to load; should be either instrument or aggregate
  * @returns {Promise<InstrumentBase|AggregateBase|null>}
  */
-export function lazyLoader (featureName, featurePart) {
-  if (featurePart === 'aggregate') {
-    switch (featureName) {
-      case FEATURE_NAMES.ajax:
-        return import(/* webpackChunkName: "ajax-aggregate" */ '../ajax/aggregate')
-      case FEATURE_NAMES.jserrors:
-        return import(/* webpackChunkName: "jserrors-aggregate" */ '../jserrors/aggregate')
-      case FEATURE_NAMES.metrics:
-        return import(/* webpackChunkName: "metrics-aggregate" */ '../metrics/aggregate')
-      case FEATURE_NAMES.pageAction:
-        return import(/* webpackChunkName: "page_action-aggregate" */ '../page_action/aggregate')
-      case FEATURE_NAMES.pageViewEvent:
-        return import(/* webpackChunkName: "page_view_event-aggregate" */ '../page_view_event/aggregate')
-      case FEATURE_NAMES.pageViewTiming:
-        return import(/* webpackChunkName: "page_view_timing-aggregate" */ '../page_view_timing/aggregate')
-      case FEATURE_NAMES.sessionTrace:
-        return import(/* webpackChunkName: "session_trace-aggregate" */ '../session_trace/aggregate')
-      case FEATURE_NAMES.spa:
-        return import(/* webpackChunkName: "spa-aggregate" */ '../spa/aggregate')
-      default:
-        throw new Error(`Attempted to load unsupported agent feature: ${featureName} ${featurePart}`)
-    }
+export function lazyLoader (agentIdentifier, featureName) {
+  checkSession(agentIdentifier)
```

Comment:
Why do we need to check session in the lazy loader? Should checking session be done in the agent class? Should that be a function of the agent class?

Reply:
We could load this in a different async module if needed, not sure what the benefits of that are